android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -576,11 +580,32 @@ constructor(
       defaultRepository.delete(resourceType = resourceType, resourceId = resourceIdentifier)
     }
   }
+  suspend fun updateResourceLastUpdatedLinkedAsSubject(
+    questionnaireResponse: QuestionnaireResponse
+  ) {
+    if (questionnaireResponse.hasSubject() && questionnaireResponse.subject.hasReference()) {
+      val resourceId = questionnaireResponse.subject.reference.extractLogicalIdUuid()
+      val resourceType =
+        questionnaireResponse
+          .subject
+          .extractType()
+          .toString()
+          .resourceClassType()
+          .newInstance()
+          .resourceType
+      resourceType.let {
+        if (it.isIn(ResourceType.Patient, ResourceType.Group)) {
+          defaultRepository.loadResource(resourceId, resourceType).let { resource ->
+            resource.updateLastUpdated()
+            defaultRepository.addOrUpdate(true, resource)
+          }
+        }
+      }
```

Comment:
no need for a `let` since you've already declared the `val`, or you could not declare the `val` and use `it` for `resourceType` everywhere in the let
```suggestion
      if (resourceType.isIn(ResourceType.Patient, ResourceType.Group)) {
        defaultRepository.loadResource(resourceId, resourceType).let { resource ->
          resource.updateLastUpdated()
          defaultRepository.addOrUpdate(true, resource)
        }
      }
```